### ISSUE TITLE
[DOXIASITETOOLS-92] Upgrade Velocity from 1.5 to 1.7

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia-sitetools#381]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[DOXIASITETOOLS-92](https://issues.apache.org/jira/browse/DOXIASITETOOLS-92?redirect=false)** and commented

Velocity 1.5 was released in 2007: http://velocity.apache.org/engine/devel/changes-report.html#a1.5

1.6.4 was done in 2010 http://velocity.apache.org/engine/devel/changes-report.html#a1.6.4

I don't have precise feature in mind, just Velocity Tools 2.0 annoucement tells Velocity 1.6 is required (seems to work well with 1.5...): http://velocity.apache.org/news.html#tools20

> This should be useable as a drop in replacement for Tools 1.4 or Tools 2.0-beta4, with a few minor exceptions. The 2.x series of VelocityTools requires Velocity 1.6 and JDK 1.5+.

From a first test, actual Doxia templates work with 1.6 but not with 1.7, so perhaps upgrading to 1.6 is a good step before trying latest 1.7

---

**Affects:** 1.6

**Issue Links:**
- [DOXIASITETOOLS-179](https://issues.apache.org/jira/browse/DOXIASITETOOLS-179) Report error line and column from Velocity runtime exception

- [MSITE-762](https://issues.apache.org/jira/browse/MSITE-762) Upgrade Velocity from 1.5 to 1.7
 (_**"is depended upon by"**_)
- [DOXIASITETOOLS-93](https://issues.apache.org/jira/browse/DOXIASITETOOLS-93) request-scoped default Velocity Tools are not accessible
 (_**"is depended upon by"**_)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

tested locally: upgrading to 1.6.4 causes failures in some maven-site-plugin ITs

examples:

```
see <<<`{/plugins/maven-site-plugin/apidocs/org/apache/maven/plugins/site/AbstractSiteRenderingMojo.html#createSiteRenderingContext(java.util.Locale)}AbstractSiteRenderingMojo.createSiteRenderingContext(...)`>>>
```

(from `index.apt.vm` in full-reporting IT) cause

```
org.apache.velocity.exception.ParseErrorException: Lexical error: org.apache.velocity.runtime.parser.TokenMgrError: Lexical error at line 100, column 141.  Encountered: "u" (117), after : "."
```

or from MSITE-609 IT, the failure happens inside the skin template:

```
[INFO] Rendering site with org.apache.maven.skins:maven-stylus-skin:jar:1.2 skin.
[ERROR] Parser Exception: META-INF/maven/site.vm
[ERROR] org.apache.velocity.runtime.parser.ParseException: Encountered "/" at line 39, column 51.
Was expecting one of:
    "," ...
    ")" ...
    <WHITESPACE> ...
```

seems we are stuck with old 1.5 until we have good explanations on upgrading instructions for users...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

No we are not, I have fixed all ITs inline but one which does not make sense.

I have upgraded Doxia Sitetools to Velocity 1.7 and Maven Site Plugin to 1.7 and ran all ITs:

1. We need to remove all references to the Maven Stylus Skin in all `site.xml` because we don't support it anymore. Several failed tests now pass.
2. The last remaining failing IT is `site-jar` which says "File[images/h3.jpg] not found in jar archive", since neither Default nor Fluido skin contain this file but only Stylus, we can remove this file from the verify script and all is fine.

If you agree, I will assign this issue to me and proceed as described.
